apps/omg_child_chain/lib/omg_child_chain/release_tasks/set_tracer.ex

```diff
@@ -14,22 +14,41 @@
 
 defmodule OMG.ChildChain.ReleaseTasks.SetTracer do
   @moduledoc false
-  use Distillery.Releases.Config.Provider
+  @behaviour Config.Provider
   require Logger
+  alias OMG.ChildChain.Tracer
   @app :omg_child_chain
 
-  @impl Provider
-  def init(_args) do
-    _ = Application.ensure_all_started(:logger)
-    config = Application.get_env(@app, OMG.ChildChain.Tracer)
-    config = Keyword.put(config, :disabled?, get_dd_disabled())
-    config = Keyword.put(config, :env, get_app_env())
+  def init(args) do
+    args
+  end
+
+  def load(config, args) do
+    _ = on_load()
+    adapter = Keyword.get(args, :system_adapter, System)
+    _ = Process.put(:system_adapter, adapter)
```

Comment:
the concern is valid. the approach here is indeed tapping into the `shared global state`. not in terms of tests though, but in terms of execution of config providers (all of them all executed by one process).

in this case, I was trying out how a direct dependency injection would work (because, if I ass the system_adapter as `args`, I would need to carry that argument through all the underlying functions)